### ISSUE TITLE
[postgres] Out of memory error

### ISSUE THREAD

**Issue by quinnhornblow** [fal-ai/fal#461]
**Describe the bug**
Using fal to write a model df with size [3627697 rows x 7 columns] returns a memory error.
```python
psycopg2.errors.OutOfMemory: out of memory
DETAIL:  Failed on request of size 262144 in memory context "ErrorContext".
```

This looks like a psycopg2/sqlalchemy error rather than a fal error but I have not had this error before when writing a dataframe of this size. Is it possible that the write method fal uses could be adjusted to fix this? Previously I have used the function below with success on a dataframe of this size.

```python
from sqlalchemy import create_engine

def push_to_db(df, table_name, schema=None, if_exists='replace'):
    """
    Push dataframe to database.
    """
    conn = None
    try:
        # connect to the PostgreSQL database
        conn = "postgresql+psycopg2://%s:%s@%s:5432/%s" % (
            SQL_USER,
            SQL_PASS,
            SQL_HOST,
            SQL_DB
        )
        engine = create_engine(conn)

        df.to_sql(
            name=table_name,
            con=engine,
            schema=schema,
            if_exists=if_exists,
            index=False,
        )
    finally:
        if conn is not None:
            engine.dispose()
            conn = None
```

**Your environment**
- OS: windows
```
fal --version 0.3.4
dbt --version 1.1.0
```
- Adapter being used: postgres

**How to reproduce**
Use fal write_to_model with a large data table

**Expected behavior**
write_to_model should succeed with large tables

**Actual behavior**
psycopg2 out of memory error

**Comment by chamini2** [comment]
Hello @quinnhornblow , thanks for reaching out. I will take a look at this ASAP. Thanks for the example script that would work for you. I will try to make it run out of memory for me and then test your script to see the difference.

**Comment by quinnhornblow** [comment]
Thanks @chamini2, using the pandas to_sql method might also be a fix for the writing datetime issue.

**Comment by chamini2** [comment]
Hello, @quinnhornblow! I think this may be better for Postgres specifically with the latest release [v0.6.0](https://pypi.org/project/fal/0.6.0/).

Can you please let me know if it is better now?

**Comment by quinnhornblow** [comment]
Fixed with version 0.6 release
